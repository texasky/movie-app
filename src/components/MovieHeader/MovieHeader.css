.movie-header {
    display: flex;
    justify-content: space-between;
    padding: 10px 5px 20px 5px;
    background-color: cornflowerblue;
}