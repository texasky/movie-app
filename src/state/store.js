import { applyMiddleware, createStore } from "redux"
import configureStore from 'redux-mock-store'
import reducers from "./reducers/index"
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

const store = createStore(reducers, composeWithDevTools(applyMiddleware(thunk)));

export default store; 