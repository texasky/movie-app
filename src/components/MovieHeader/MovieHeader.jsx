import MovieAppLogo from '../MovieAppLogo/MovieAppLogo';
import MovieSearchBar from '../MovieSearchBar/MovieSearchBar';
import './MovieHeader.css';

const MovieHeader = (props) => {

    return (
        <div className='movie-header'>
            <div className='movie-header-wrapper'>
                <MovieAppLogo/>
                <MovieSearchBar/>
            </div>
        </div>
    )
}

export default MovieHeader;