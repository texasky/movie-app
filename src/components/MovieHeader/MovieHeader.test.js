import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MovieHeader from './MovieHeader';
import { Provider } from 'react-redux';
import store from './../../state/store'

test('Movi header component tests', () => {
    const utils = render(<Provider store={store}><MovieHeader /></Provider>);

    const movieHeader = utils.getByTestId('movie-header');

    expect(movieHeader).toBeInTheDocument();
});

test('Movi header component matches snapshot', () => {
    const tree = renderer.create(<Provider store={store}><MovieHeader /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});