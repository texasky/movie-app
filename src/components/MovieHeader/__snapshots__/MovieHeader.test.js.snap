// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movi header component matches snapshot 1`] = `
<div
  className="movie-header"
  data-testid="movie-header"
>
  <div
    className="movie-header-wrapper"
  >
    <div
      className="movie-app-logo"
      data-testid="movie-app-logo"
    >
      <img
        alt="MOVIE APP"
        height={40}
        src="movie-app-icon.jpeg"
        width={40}
      />
      <span>
        MOVIE APP. TYPE AND SEARCH :)
      </span>
    </div>
    <div
      className="movie-search-bar"
    >
      <input
        aria-label="search-input"
        data-testid="movie-search-input"
        disabled={false}
        id="movie-search-input"
        onChange={[Function]}
        placeholder="Type to search"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
